### PR TITLE
Feature: Platform view-only users

### PR DESCRIPTION
**Pull request by petedejoy**
This PR addresses the issue summarized here: https://github.com/astronomer/issues/issues/552

We would like to, in certain Enterprise cases, limit all users from creating workspaces on the platform. Since some of our customers will have hundreds of users, it is a security concern for them to give users the ability to consume Kubernetes resources via creating one-off Airflow environments.

With this PR, all users are now given a synthetic `USER` role, with a permission `system.workspace.create` attached to it. The UX will not change by default, but folks installing Astronomer Enterprise can now move that permission onto a `SYSTEM_*` role if they would like to configure the platform such that _only_ users with system-level roles can create new workspaces.

### CODE REVIEW

File: src/lib/rbac/index.js [hunk 1]
```diff
@@ -178,12 +178,10 @@ export async function getAuthUser(authorization) {
  * This function wraps the two calls above to append fake roleBindings
  * to the user object for any deployments that belong to workspaces where
  * the user has WORKSPACE_ADMIN role.
- * @param {Promise} promise A proimse for a user or service account.
+ * @param {User} Resolved promise for a user or service account.
```
Comment (ashb) [inline src/lib/rbac/index.js]:
```suggestion
 * @param {User} The user or service account object.
```

**Comment by petedejoy** [comment]
Yeah I was thinking that as well- I didn't see a test in there for `addDeploymentRoleBindings` either, is that just because we're planning on deprecating it with deployment-level RBAC?

Also, I confirmed in Orbit that the feature works, but we probably need to couple this PR with some additional error-handling in the UI. I think this works to merge as an MVP for customers, but will plan on leaving the issue open until I can get a better UX implemented in Orbit

image

**Comment by petedejoy** [comment]
Added some unit tests for our new `addUserRoleBinding` function. Happy to do the same for the existing `addDeploymentRoleBindings` function if you all think it'd be worth it!

File: config/default.yaml [hunk 2]
```diff
@@ -245,6 +245,12 @@ roles:
       ? system.workspace.suspend
       ? system.workspace.extendTrial
       ? system.airflow.admin
+      ? system.workspace.create
+    
+  USER:
+    name: User
+    permissions:
+
```
Comment (schnie) [inline config/default.yaml]: @petedejoy don't we want normal users to be able to create workspaces by default?

Comment (petedejoy) [reply config/default.yaml]: Ah yeah sorry, changed this to test and forgot to revert back.

File: config/default.yaml [hunk 3]
```diff
@@ -245,6 +245,11 @@ roles:
       ? system.workspace.suspend
       ? system.workspace.extendTrial
       ? system.airflow.admin
+    
+  USER:
```
Comment (schnie) [inline config/default.yaml]: @petedejoy can we bump this to the bottom of the list here so it doesn't break up the scoped roles.

File: database/datamodel.graphql [hunk 4]
```diff
@@ -2,6 +2,7 @@ enum Role {
   SYSTEM_ADMIN
   SYSTEM_EDITOR
   SYSTEM_VIEWER
+  USER
```
Comment (schnie) [inline database/datamodel.graphql]: and same here

File: config/default.yaml (hunk 3 above)
Comment (petedejoy) [reply config/default.yaml]: Whoops, done!

File: config/default.yaml [hunk 5]
```diff
@@ -245,11 +245,10 @@ roles:
       ? system.workspace.suspend
       ? system.workspace.extendTrial
       ? system.airflow.admin
-
+
```
Comment (schnie) [inline config/default.yaml]:
```suggestion

```

File: config/default.yaml [hunk 6]
```diff
@@ -245,11 +245,9 @@ roles:
       ? system.workspace.suspend
       ? system.workspace.extendTrial
       ? system.airflow.admin
```
Comment (schnie) [inline config/default.yaml]:
```suggestion
      ? system.airflow.admin

```

**Pull request merged**